mimic/model/markov_chain_model.py

```diff
@@ -1,8 +1,46 @@
 """Markov chain model class."""
-from mimic.model.model import Model
 
+import random
+from mimic.model.model import Model
 
 class MarkovChainModel(Model):
-    """ML model utilizing Markov chain."""
+    """A type of model"""
+
+    def __init__ (self, stateLength):
+        self.order = stateLength
+        self.groupSize = stateLength + 1
+        self.dict = {}
+        self.data = None
+
+    def learn(self, data):
+        """Takes in a list of words as an argument
+        and constructs a dictionary based
+        on stateLength provided by the user"""
+        print('Learning...')
+        self.data = data.split()
+
+        for i in range(0,len(data) - self.groupSize):
+            key = tuple(data[i: i +self.order])
+            value = data[i+ self.order]
+
+            if key in self.dict:
```

Comment:
In Python, this type of behaviour can be improved with a [`defaultdict`](https://docs.python.org/2/library/collections.html) — if you initialize `self.dict = defaultdict(list)`, you can just do `self.dict[key].append(value)` regardless of whether `key` is in `self.dict`, and in the case that it's the first insertion it will automatically initialize the list.

Reply:
Implemented these changes, thanks!